### PR TITLE
Add basic string functions to templates

### PR DESCRIPTION
**Pull request by n-i-x**
## Description

Adds the ability to add functions to templates. Currently supports modifying a variable to all uppercase or all lowercase.

Usage:

{{.ProviderName | ToLower}} {{.ProviderName | ToUpper}}

## Motivation and Context

I designed a template for oauth2-proxy which dynamically displays the provider login button based on the provider name using standard external CSS libraries. The required class names are btn-{{lowercaseProvider}}. Rather than create new variables to pass in to the template, the better solution seemed to be to allow the template engine to use basic string functions like ToUpper and ToLower.

HTML templates using external libraries may need to use the provider name in a class. As classes generally cannot be renamed unless a pre-processor like SASS or LESS is used, modifying the variable in the template is the best option.

## How Has This Been Tested?

A template has been created which requires lowercase class names for the provider to display the provider logo using font-awesome.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [ ] I have updated the documentation/CHANGELOG accordingly.
- [ ] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: templates.go [hunk 1]
```diff
@@ -12,7 +13,12 @@ func loadTemplates(dir string) *template.Template {
 		return getTemplates()
 	}
 	logger.Printf("using custom template directory %q", dir)
-	t, err := template.New("").ParseFiles(path.Join(dir, "sign_in.html"), path.Join(dir, "error.html"))
+	funcMap := template.FuncMap{
+		"ToUpper": strings.ToUpper,
+		"ToLower": strings.ToLower,
+	}
+	logger.Printf("using custom template directory %q", dir)
```
Comment (JoelSpeed) [inline templates.go:20]: This log line is duplicated with line 15

Comment (n-i-x) [reply templates.go:20]: I’ll take a stab at a test. Not sure how I missed that duplicate logging entry...

Comment (n-i-x) [reply templates.go]: Tests added... PTAL

Comment (n-i-x) [reply templates.go]: CHANGELOG entry added

Comment (n-i-x) [reply templates.go]: sorry about the lint error.. I used gitpod and my editor settings weren't configured there to format on save

**Pull request merged**